### ISSUE TITLE
`comparison-overlap` error not triggered for booleans

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#11364]
```py
from typing import Literal

foo: Literal["foo"] = "foo"

# error: Non-overlapping equality check (left operand type: "Literal['foo']", right operand type: "Literal['bar']")  [comparison-overlap]
foo == "bar"


bar: Literal[True]

# no error
bar == False
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-context%2Cshow-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=4d291b48d0f5858f4b1ec54a1a67e5f6

**Comment by JukkaL** [comment]
I think that this is by design. Bools are often used for things like debug flags that can typically be false, but it's still okay to write `if DEBUG: print('something')`. The idea is that the developer will temporarily modify the value of the flag during a debugging session if they want verbose logging.

`if DEBUG == True:` seems similar enough to `if DEBUG:` that they should probably use the same rules.

**Comment by KotlinIsland** [comment]
```py
from typing import Literal

foo: Literal["foo"]

# error: Non-overlapping equality check (left operand type: "Literal['foo']", right operand type: "Literal['bar']")  [comparison-overlap]
foo == "bar"
foo != "bar"

# no error
foo != "foo"
foo == "foo"


bar: Literal[True]

# no error
bar == True
bar == False
bar != True
bar != False
if bar: print(1)

# error: Statement is unreachable  [unreachable]
if not bar: print(1)
```
([Gist]((https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-context%2Cshow-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=4c8259b1f4bee5c198e0cc252c3fe00c))
There are a lot of simple situations that mypy doesn't seem to pick up on, all of these lines of code are suspect given the type annotations.

> Bools are often used for things like debug flags
@JukkaL Do you mean debug flags that are annotated as `Literal[False]` for purposes of compilation optimization? If not I don't really understand your point. If the debug flag is annotated as `DEBUG: Final = False` then I would that that that is incorrect and should be annotated as `Final[bool]`. If this truly is intended then it is quite non-obvious, and not documented. If there is a need for this feature(compiler optimization), then perhaps a mypyc `Annotated` flag could be implemented, which would have the benefit of being discover-able and obvious:

```py
DEBUG: Final[Annotated[bool, MYPYC_FALSE]] = False
```

I think that as it stands it's a inconsistent feature, as debug options can also be integers or strings, and the benefits of catching true `comparison-overlap` errors far outweighs this very hidden design in my opinion.

**Comment by KotlinIsland** [comment]
Researching this further, I have discovered that mypy already special cases artificially literal bools:
```py
# mypy: always-true=foo
from typing import Literal

foo: bool

if foo: print(1)  # no error

bar: Literal[True]

if not bar: print(1)  # error: Statement is unreachable
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=warn-unreachable&gist=155eb9beccbfde903e2b3a5d61f0752e
